kpatch-build/kpatch-elf.c

```diff
@@ -76,6 +76,31 @@ int is_debug_section(struct section *sec)
 	       !strncmp(name, ".eh_frame", 9);
 }
 
+#ifdef __powerpc__
+/*
+ * Number of instructions between global and local entry point of a
+ * function is mostly 8 instructions. i.e st_other == 3. The count of
+ * instruction can be decoded only for local function symbols.
+ */
+unsigned int has_local_entry_offset(GElf_Sym sym)
+{
+	return !!PPC64_LOCAL_ENTRY_OFFSET(sym.st_other);
+}
+
+/*
+ * As per ELF ABI v2 specification, any function those symbol reference
+ * is to be dynamically resolved. The st_shndx member of such symbol
+ * is set to SHN_UNDEF.
+ */
+unsigned int is_local_func(struct symbol *sym)
+{
+	if (sym->type != STT_FUNC || sym->sym.st_shndx == SHN_UNDEF)
+		return 0;
+
+	return 1;
```

Comment:
I think removing the double negative would be more readable:
```
return (sym->type == STT_FUNC && sym->sym.st_shndx != SHN_UNDEF);
```
Also, why not check `sym->bind != STB_GLOBAL` instead of `sym->sym.st_shndx != SHN_UNDEF`?